src/panels/lovelace/cards/hui-map-card.js

```diff
@@ -0,0 +1,258 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+import Leaflet from 'leaflet';
+
+import '../../map/ha-entity-marker.js';
+
+import processConfigEntities from '../common/process-config-entities';
+import computeStateDomain from '../../../common/entity/compute_state_domain.js';
+import computeStateName from '../../../common/entity/compute_state_name.js';
+
+Leaflet.Icon.Default.imagePath = '/static/images/leaflet';
+
+class HuiMapCard extends PolymerElement {
+  static get template() {
+    return html`
+      <style>
+        ha-card:not(.panel) {
+          line-height: 0;
+          overflow: hidden;
+        }
+        
+        ha-card.panel {
+          left: 0;
+          top: 0;
+          width: 100%;
+          /**
+           * In panel mode we want a full height map. Since parent #view 
+           * only sets min-height, we need absolute positioning here
+           */
+          height: 100%;
+          position: absolute;
+        }
+        
+        #map {
+          z-index: 0;
+          border: none;
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 100%;
+          height: 100%;
+        }
+        
+        #root {
+          width: 100%;
+          position: relative;
+        }
+        
+        ha-card.panel > #root {
+          height: 100%;
+        }
+      </style>
+      
+      <ha-card id="card" header="[[_config.title]]">
+        <div id="root">
+          <div id="map"></div>
+        </div>
+      </ha-card>
+
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: Object,
+      _config: Object,
+      isPanel: Boolean
```

Comment:
If you add reflectToAttribute: true, you can use css like `:host([is-panel]) #root`